platform/linux/mod.rs

```diff
@@ -278,16 +286,10 @@ impl UnixSender {
 
                 if result <= 0 {
                     let error = UnixError::last();
-                    if error.0 == libc::ENOBUFS && bytes_to_send > 2000 {
+                    if error.0 == libc::ENOBUFS
+                       && downsize(&mut sendbuf_size, bytes_to_send).is_ok() {
```

Comment:
This does not handle the Err(()) case from downsize correctly - perhaps this should be:
```rust
if error.0 == libc::ENOBUFS {
    // If the kernel failed to allocate a buffer large enough for the packet,
    // retry with a smaller size (if possible).
    try!(downsize(&mut sendbuf_size, bytes_to_send));
    continue;
} else {
```

Note this may need a tweak for freeing behavior.

There is a similar issue above, but a later commit makes it moot.